homeassistant/components/notify/rocketchat.py

```diff
@@ -0,0 +1,69 @@
+"""
+Rocket.Chat notification service.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/notify.rocketchat/
+"""
+import asyncio
+import logging
+
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_URL, CONF_USERNAME, CONF_PASSWORD)
+from homeassistant.components.notify import (
+    ATTR_DATA, PLATFORM_SCHEMA,
+    BaseNotificationService)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['rocketchat-API==0.6.1']
+
+CONF_ROOM = 'room'
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_URL): cv.string,
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Required(CONF_ROOM): cv.string,
+})
+
+
+@asyncio.coroutine
+def async_get_service(hass, config, discovery_info=None):
+    """Return the notify service."""
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+
+    url = config.get(CONF_URL)
+    room = config.get(CONF_ROOM)
+
+    return RocketChatNotificationService(hass, url, username, password, room)
```

Comment:
Should this still be returned if the connection fails in the instance init?

Reply:
No, it probably shouldn't. What is the preferred way to handle this? 1. Move the existing exception handling here (allowing the constructor to throw those exceptions, 2. Throw a different exception from the constructor (one provided by HASS?).